### PR TITLE
Added code for using the Postmark template system

### PR DESCRIPTION
**Pull request by cwurld**
Added code to the core.PMMail class to make it easy to use the Postmark template system. Updated validation code to handle new kwargs. Updated README.md.

### CODE REVIEW

**Comment by nicholasserra** [comment]
Thanks for the feature :) We'd like to get some unit tests around new code, do you think that would be possible with this change? I'll spend some time reviewing this later this week.

**Comment by cwurld** [comment]
Yes, I will write some unit tests. Give me a few days.

File: postmark/core.py [hunk 1]
```diff
@@ -68,14 +68,17 @@ def __init__(self, **kwargs):
...
                         Can be multiple values separated by commas (limit 20)
-        subject:        Subject of the email
+        subject:        Subject of the email. ***DO NOT SET if using Postmark templates
         tag:            Use for adding categorizations to your email
         html_body:      Email message in HTML
         text_body:      Email message in plain text
         track_opens:    Whether or not to track if emails were opened or not
         custom_headers: A dictionary of key-value pairs of custom headers.
         attachments:    A list of tuples or email.mime.base.MIMEBase objects
+        attachments:    A list of tuples or email.mime.base.MIMEBase objects
```
Comment (butorov) [inline postmark/core.py]: I think this line is duplicated by mistake.

Comment (cwurld) [inline postmark/core.py]: Good catch. I removed it the most recent commit.

**Comment by cwurld** [comment]
I added tests to the latest commit.

File: postmark/core.py [hunk 2]
```diff
@@ -338,9 +345,13 @@ def _check_values(self):
             raise PMMailMissingValueException('Cannot send an e-mail without a sender (.sender field)')
         elif not self.__to:
             raise PMMailMissingValueException('Cannot send an e-mail without at least one recipient (.to field)')
-        elif not self.__subject:
+        elif bool(self.__template_id) != bool(self.__template_model):
```
Comment (nicholasserra) [inline postmark/core.py]: Had to think about what was happening here for a bit. I think this reads easier as:

```
elif (self.__template_id or self.__template_model) and not all([self.__template_id or self.__template_model]):
```

File: postmark/core.py [hunk 3]
```diff
@@ -402,11 +419,14 @@ def to_json_message(self):
...
 
-    def send(self, test=None):
+    def send(self, endpoint=None, test=None):
         '''
         Send the email through the Postmark system.
         Pass test=True to just print out the resulting
         JSON message being sent to Postmark
+
+        endpoint: use this to post to other Postmark API endpoints. If not set, the default is "email"
+
```
Comment (nicholasserra) [inline postmark/core.py]: Where is this default `email` coming in? Shouldn't it be in the function declaration?

File: postmark/core.py (hunk 2 above)
Comment (cwurld) [inline postmark/core.py]: Yes, that was a new idiom to me as well. I found it on stackoverflow and though it was pretty cool, so I went with it. But I am happy to change.

What's the best way to change it? Should I change it in my repo? Or do you want to change it after the merge?

Comment (nicholasserra) [inline postmark/core.py]: Change it in this PR please :D

Comment (cwurld) [inline postmark/core.py]: Opps. I changed that code when I added tests. Do I need to do a new pull request?

Comment (nicholasserra) [inline postmark/core.py]: If you push to this branch it will change the PR.

Comment (cwurld) [inline postmark/core.py]: done

File: postmark/core.py (hunk 3 above)
Comment (cwurld) [inline postmark/core.py]: Yes. I changed the default in the function declaration.

**Comment by nicholasserra** [comment]
This is looking pretty good. Will need newest changes from master pulled and merged in, or rebased.

**Comment by graphiclife** [comment]
👍

File: README.md [hunk 4]
```diff
@@ -96,6 +96,55 @@ class EmailHandler(tornado.web.RequestHandler, PostmarkMixin):
...
+            'description': 'Legal Services',
+            'amount': '100.00'
+        }
+    ],
+    'opening_balance': '2000.00',
+    'closing_balance': "<span class='negative_amount'>1800.00</span>",
+    'notes': 'Thank you for choosing Acme Law.',
+}
+pm = PMMail(to='groucho.marx@gmail.com', sender='harpo.marx@gmail.com',
+            template_id=123456, template_model=template_context)
```
Comment (ianwalter) [inline README.md]: Where is template_context coming from? Shouldn't it be `template_model=template_model`?

**Comment by ianwalter** [comment]
@cwurld It's a shame this seems to be 99% done but has been dormant for over a year.

**Comment by cwurld** [comment]
I fixed the template_model/template_context issue in README.md as pointed out by @ianwalter,  and did a rebase.

**Comment by nicholasserra** [comment]
Did you mean to close this?

**Comment by cwurld** [comment]
Opps. No.

**Comment by nicholasserra** [comment]
Thank you for the work on this :)

**Pull request merged**